src/App.vue

```diff
@@ -1,31 +1,37 @@
 <template>
   <div id="app">
-    <custom-header />
-    <result-control :number_of_movies="1" />
-    <!-- <film-gallery /> -->
-    <custom-footer company_name="netflix" app_name="roulette" />
+    <search-page v-if="current_page == 'search-page'" />
+    <card-details-page v-if="current_page == 'card-details-page'" :card="card" />
   </div>
 </template>
 
 <script>
 import "./assets/app.css";
-import CustomHeader from './components/CustomHeader.vue';
-import CustomFooter from './components/CustomFooter.vue';
-// import FilmGallery from './components/FilmGallery.vue';
-import ResultControl from './components/ResultControl.vue';
+import CardDetailsPage from './components/CardDetailsPage.vue';
+import SearchPage from './components/SearchPage.vue';
+import { EventBus } from "./event-bus";
 
 export default {
   name: "App",
   components: {
-    CustomHeader,
-    CustomFooter,
-    ResultControl,
-    // FilmGallery
+    SearchPage,
+    CardDetailsPage
   },
-  methods: {
-    getMovies: function() {
-      
-    }
+  mounted() {
+    EventBus.$on("changeToSearchPage", () => {
```

Comment:
Consider creating a "core" folder under "src" where all the main reusable non-react things will be stored.

For this particular line:
Under "src/core" folder that you create after one of the above comments consider creating a folder "constants" that will hold any number of file to your liking - as long as they contain all the hard-coded numbers/strings that you use throughout the app. 
E.g. "changeToSearchPage" event string, "search-page", "card-details-page", etc.The idea is to have a central place for all the constants to be able to easily reuse/change them.